src/ln/channelmonitor.rs

```diff
@@ -1903,6 +1919,13 @@ impl ChannelMonitor {
 								// has timed out, or we screwed up. In any case, we should now
 								// resolve the source HTLC with the original sender.
 								payment_data = Some(((*source).clone(), htlc_output.payment_hash));
+							} else if !$local_tx {
```

Comment:
Note that we still want to hit the last else even if !$local_tx as long as payment_data is still none.